### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by jadx2**
Implement the bubble_sort_by method with the use of yield.

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      if arr[i] > arr[i + 1]
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swapped = true
+      end
+    end
+
+    break unless swapped
+  end
+  puts arr
+end
```
Comment (arslanbisharat) [inline bubble_sort.rb:16]: - [ ] bubble_sort should sort the array of numbers in increasing order. 
For Example: `If array = [1, 3, 5, 4, 2] then sorted_array = [1, 2, 3, 4, 5]`

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        if yield(left, right).positive?
+          arr[i], arr[i + 1] = arr[i + 1], arr[i]
+          swapped = true
+        end
+      end
+      break unless swapped
+    end
+    puts arr
+  else
+    puts 'You need to add a block'
```
Comment (arslanbisharat) [inline bubble_sort.rb:38]: - [ ] bubble_sort_by should sort the array by accepting a block. 


- [ ] bubble_sort_by should raise the error if the block is not given

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Request changes 😳

I am very happy to see this project and I can see you worked hard on this project. There are only a few issues that we need to address and I am sure that you will end up with a perfect project

_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_

This is a great opportunity to learn. Microverse is here to help you. 

**Please have a look at the suggestions to get more understanding.**

- [ ] Your code is not working as it is supposed to work. I have given suggestions below. Please work on them. 


**Please make the changes and request a re-review for the given project**

If you get stuck somewhere, don't hesitate to reach me out or schedule s 1:1 meeting with a Full-time TSE. 

Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

File: bubble_sort.rb (hunk 2 above)
Comment (jadx2) [reply bubble_sort.rb:38]: Could it be because it is not returning the result but puts?

**Review by jubaan — APPROVED** [review]
# :tada: :tada: :tada: :rotating_light: Status: Approved :rotating_light:  :tada: :tada: :tada:

#### :clap: :clap: Great job :hammer_and_pick:!! There are no more issues left to be addressed. 

## See you at the top :rocket:!

# :tada: :tada: :tada: :rotating_light: Status: Approved :rotating_light:  :tada: :tada: :tada:

**Pull request merged**